hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -392,6 +398,12 @@ public void initTableMetadata() {
     }
 
     if (!exists) {
+      if (metadataWriteConfig.isMetadataAsyncIndex()) {
```

Comment:
guess we missed one flow here. 
lets say someone brought down every writer. And starts HoodieIndexer for the first time by enabling all partitions with the intention that everything will get built out. I see we come to this code path and eventually call scheduleIndex in L404, including FILES partition. Guess the plan is to always follow synchronous code path for FILES partition.

Reply:
likely the fix is getEnabledPartitionTypes should not have the FILES partition.

Reply:
will remove this conditional.. it is not needed here.